module_utils/network_lsr/utils.py

```diff
@@ -126,6 +126,29 @@ def GMainLoop_iterate_all(cls):
     def create_cancellable(cls):
         return cls.Gio().Cancellable.new()
 
+    @classmethod
+    def create_callback(cls, finish_method):
+        """
+        Create a callback that will return the result of the finish method and
+        quit the GMainLoop
+
+        :param finish_method str: Name of the finish method to call from the
+        source object in the callback
+        """
+
+        def callback(source_object, res, user_data):
+            success = None
+            try:
+                success = getattr(source_object, finish_method)(res)
+            except Exception as e:
+                if Util.error_is_cancelled(e):
+                    return
+                user_data["error"] = str(e)
+            user_data["success"] = success
+            Util.GMainLoop().quit()
```

Comment:
`create_callback()` is a class method. I think that is preferable over a static-method if the method references itself (its class). But here you still use `Util` directly. Either make the method `@staticmethod` or use `cls` instead of `Util`. I prefer the latter, because I find it a bit ugly that a static-method would access itself again via the `Util` name.

Reply:
good catch, will use `cls`.